src/OrchardCore.Modules/OrchardCore.Localization/Startup.cs

```diff
@@ -48,13 +48,13 @@ public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder ro
             var localizationOptions = serviceProvider.GetService<IOptions<RequestLocalizationOptions>>().Value;
             var cultureOptions = serviceProvider.GetService<IOptions<CultureOptions>>().Value;
 
-            var requestLocalizationOptions = new OrchardCoreRequestLocalizationOptions(ignoreSystemSettings: cultureOptions.IgnoreSystemSettings)
-                .WithRequestLocalizationOptions(localizationOptions)
+            var localizationOptionsBuilder = new RequestLocalizationOptionsBuilder(localizationOptions, !cultureOptions.IgnoreSystemSettings);
+            localizationOptionsBuilder
                 .SetDefaultCulture(defaultCulture)
                 .AddSupportedCultures(supportedCultures)
                 .AddSupportedUICultures(supportedCultures);
 
-            app.UseRequestLocalization(requestLocalizationOptions);
+            app.UseRequestLocalization(localizationOptionsBuilder.Options);
```

Comment:
Normally a builder has a `Build()` method that here would just return the options it has mutated, so I would first suggest something like this if you implement the `.Build()` and update other methods as suggested.

    localizationOptions = new RequestLocalizationOptionsBuilder(localizationOptions, !cultureOptions ...)
        .SetDefaultCulture(defaultCulture)
        .AddSupportedCultures(supportedCultures)
        .AddSupportedUICultures(supportedCultures)
        .Build();

    app.UseRequestLocalization(localizationOptions);

If you don't want to implement a `.Build()`, yes more like an updater than a builder (anyway will be removed in a future version), because it acts on the options instance we pass to it, you could just do the following, and then maybe rename it to `RequestLocalizationOptionsUpdater`.

    new RequestLocalizationOptionsUpdater(localizationOptions, !cultureOptions ...)
        .SetDefaultCulture(defaultCulture)
        .AddSupportedCultures(supportedCultures)
        .AddSupportedUICultures(supportedCultures);

    app.UseRequestLocalization(localizationOptions);

Reply:
> Normally a builder has a Build()

Not necessary, please hold on until I finish few things